internal/pkg/agent/perms/windows.go

```diff
@@ -83,52 +92,41 @@ func FixPermissions(topPath string, opts ...OptFunc) error {
 		// token to have the 'SeRestorePrivilege' or it's unable to adjust the ownership
 		return winio.RunWithPrivileges([]string{winio.SeRestorePrivilege}, func() error {
 			return filepath.WalkDir(topPath, func(name string, _ fs.DirEntry, err error) error {
-				if err == nil {
+				switch {
+				case err == nil:
 					// first level doesn't inherit
-					inherit := true
-					if topPath == name {
-						inherit = false
-					}
+					inherit := topPath == name
 
-					err = acl.Apply(name, true, inherit, grants...)
-					if err != nil {
-						// Check for Errno = 0 which indicates success
-						// https://pkg.go.dev/golang.org/x/sys/windows#Errno
-						if errors.Is(err, syscall.Errno(0)) {
-							return nil
```

Comment:
watch here, previously if this was syscall.Errno(0), we were returning which IMO shouldn't be the case, and we should continue below trying to take onwership